builtin/mainmenu/tab_online.lua

```diff
@@ -331,6 +336,14 @@ local function main_button_handler(tabview, fields, name, tabdata)
 		end
 	end
 
+	if fields.btn_add_favorite then
+		local server = find_selected_server()
+
+		serverlistmgr.add_favorite(server)
```

Comment:
```suggestion
		serverlistmgr.add_favorite(find_selected_server())
```
(and then get rid of the `local server` right above)